test-network-function/lifecycle/suite.go

```diff
@@ -176,45 +156,26 @@ func testScaling(env *config.TestEnvironment) {
 	ginkgo.It(testID, func() {
 		ginkgo.By("Testing deployment scaling")
 		defer results.RecordResult(identifiers.TestScalingIdentifier)
+		defer restoreDeployments(env)
 
-		namespaceDeploymentsBackup := make(map[string]dp.DeploymentMap)
-		defer restoreDeployments(env, &namespaceDeploymentsBackup)
-
-		// Map to register the deployments that have been already tested
-		deploymentNames := make(map[string]bool)
-
-		for _, podUnderTest := range env.PodsUnderTest {
-			podName := podUnderTest.Name
-			namespace := podUnderTest.Namespace
-
-			// Get deployment name and check whether it was already tested.
-			// ToDo: Proper way (helper/handler) to do this.
-			podNameParts := strings.Split(podName, "-")
-			deploymentName := podNameParts[0]
-			msg := fmt.Sprintf("Testing deployment=%s, namespace=%s pod name=%s", deploymentName, namespace, podName)
-			log.Info(msg)
-			if _, alreadyTested := deploymentNames[deploymentName]; alreadyTested {
-				continue
-			}
+		if len(env.DeploymentsUnderTest) == 0 {
+			ginkgo.Fail("No test deployments found.")
+		}
 
-			// Save deployment data for deferred restoring in case something's wrong during the TC.
-			deployments, _ := getDeployments(namespace)
-			deployment := deployments[deploymentName]
-			saveDeployment(namespaceDeploymentsBackup, namespace, deploymentName, &deployment)
+		for _, deployment := range env.DeploymentsUnderTest {
+			ginkgo.By(fmt.Sprintf("Scaling Deployment=%s, Replicas=%d (ns=%s)",
+				deployment.Name, deployment.Replicas, deployment.Namespace))
 
 			replicaCount := deployment.Replicas
 
 			// ScaleIn, removing one pod from the replicaCount
-			runScalingTest(namespace, deploymentName, (replicaCount - 1))
+			runScalingTest(deployment.Namespace, deployment.Name, (replicaCount - 1))
```

Comment:
you can pass deployment only as param here, then get the two variables from deployment. 
I don't think we will ever need to test a deployment1.name with deployment2.Namespace

Reply:
same thing for replicaCount.

Reply:
Done.